From 4bf09cfd604caa652a13d870c9fc4c0af4b3f833 Mon Sep 17 00:00:00 2001
From: Nick Xie <nick@khadas.com>
Date: Thu, 9 Apr 2026 15:49:30 +0800
Subject: [PATCH 83/91] arm64: dts: amlogic: t7: add RTC support

Add I2C0 bus node and pinmux configuration for the Amlogic T7 SoC.
Enable I2C0 and add the Haoyu HYM8563 RTC device for Khadas VIM4.
Add rtc0 (hym8563) and rtc1 (vrtc) aliases.

Signed-off-by: Nick Xie <nick@khadas.com>
---
 .../boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts    | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts b/arch/arm64/boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts
index b4a5f5ce3e90..8ca2d0dc0730 100644
--- a/arch/arm64/boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts
+++ b/arch/arm64/boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts
@@ -18,6 +18,8 @@ aliases {
 		mmc0 = &sd_emmc_c;
 		mmc1 = &sd_emmc_b;
 		mmc2 = &sd_emmc_a;
+		rtc0 = &rtc;
+		rtc1 = &vrtc;
 	};
 
 	memory@0 {
@@ -255,6 +257,12 @@ khadas_mcu: system-controller@18 {
 		fan-supply = <&vcc5v>;
 		#cooling-cells = <2>;
 	};
+
+	rtc: rtc@51 {
+		compatible = "haoyu,hym8563";
+		reg = <0x51>;
+		#clock-cells = <0>;
+	};
 };
 
 &nna_thermal {
-- 
2.53.0

